src/reference/builder.js

```diff
@@ -1,91 +1,95 @@
 /*global Promise,  _*/
 'use strict';
 
-  /* Filename: helpers/reference_helper.js  */
-  //Dependency gestion depending on the fact that we are in the browser or in node.
-  var fetch = require('../network/fetch');
-  var checkIsString = require('../util/string/check');
+/* Filename: helpers/reference_helper.js  */
+//Dependency gestion depending on the fact that we are in the browser or in node.
+import fetch from '../network/fetch';
+import checkIsString from '../util/string/check';
 
-  //Container for the list and
-  var {getElement, getCacheDuration} = require('./config');
+//Container for the list and
+import {getElement, getCacheDuration} from './config';
 
-  let cache = {};
+let cache = {};
+const promiseWaiting = new Set();
```

Comment:
@Hartorn Ok is it essential to use a `Set`? Just in case we have projects which are on ie 10.
I agree that it is elegant and cool.